### PR TITLE
Update Development Automation and Pull Request Process

### PR DESCRIPTION
**Pull request by xiangpengzhao**
fixes: #834 

/cc @bgrant0607 @spiffxp

### CODE REVIEW

File: contributors/devel/automation.md [hunk 1]
```diff
@@ -38,37 +38,33 @@ The status of the submit-queue is [online.](http://submit-queue.k8s.io/)
...
 A PR is considered "ready for merging" if it matches the following:
-  * The PR must have the label "cla: yes" or "cla: human-approved"
-  * The PR must be mergeable. aka cannot need a rebase
-  * All of the following github statuses must be green
-     * Jenkins GCE Node e2e
-     * Jenkins GCE e2e
-     * Jenkins unit/integration
+  * The PR must have the label `cncf-cla: yes` or `cla: human-approved`
+  * The PR must be mergeable, aka cannot need a rebase
+  * All of the [Required Github CI Tests](https://github.com/kubernetes/test-infra/blob/master/mungegithub/submit-queue/deployment/kubernetes/configmap.yaml#L15) must be green
```
Comment (cblecker) [inline contributors/devel/automation.md]: This list is available in a more user friendly version at https://submit-queue.k8s.io/#/info on the Merge Requirements tab. That tab is also dynamically updated, so perhaps it's a better source for this list?

File: contributors/devel/automation.md [hunk 2]
```diff
@@ -38,37 +38,33 @@ The status of the submit-queue is [online.](http://submit-queue.k8s.io/)
...
+  * The PR must have the `lgtm` label. The `lgtm` label is automatically applied
+    following a review comment consisting of `/lgtm` (case-insensitive) in a single line
+  * The PR must not have been updated since the `lgtm` label was applied
+  * The PR must have the `approved` label. The `approved` label is automatically applied after all approvers have approved the PR 
+    by a comment consisting of `/approve` (case-insensitive) in a single line
+  * The PR must not have the `do-not-merge` label or any of other `do-not-merge/*` labels, e.g., `do-not-merge/release-note-label-needed`
 
 ### Merge process
 
 Merges _only_ occur when the [critical builds](http://submit-queue.k8s.io/#/e2e)
```
Comment (cblecker) [inline contributors/devel/automation.md]: There is no e2e tab anymore and no such thing as critical builds. All the ci builds are non-blocking.

File: contributors/devel/automation.md [hunk 3]
```diff
@@ -38,37 +38,33 @@ The status of the submit-queue is [online.](http://submit-queue.k8s.io/)
...
 are passing. We're open to including more builds here, let us know...
 
-Merges are serialized, so only a single PR is merged at a time, to ensure
-against races.
-
 If the PR has the `retest-not-required` label, it is simply merged. If the PR does
-not have this label the e2e, unit/integration, and node  tests are re-run. If these
-tests pass a second time, the PR will be merged as long as the `critical builds` are
+not have this label, the [Required Github CI Tests](https://github.com/kubernetes/test-infra/blob/master/mungegithub/submit-queue/deployment/kubernetes/configmap.yaml#L15)
```
Comment (cblecker) [inline contributors/devel/automation.md]: Same as above, the list of retests needed is on the merge requirements page

File: contributors/devel/automation.md [hunk 4]
```diff
@@ -38,37 +38,33 @@ The status of the submit-queue is [online.](http://submit-queue.k8s.io/)
...
+are re-run. If these tests pass a second time, the PR will be merged as long as the `critical builds` are
 green when this PR finishes retesting.
 
 ## Github Munger
 
 We run [github "mungers"](https://github.com/kubernetes/contrib/tree/master/mungegithub).
 
-This runs repeatedly over github pulls and issues and runs modular "mungers"
-similar to "mungedocs." The mungers include the 'submit-queue' referenced above along
+This runs repeatedly over github pulls and issues and runs modular `mungers`
```
Comment (cblecker) [inline contributors/devel/automation.md]: single backticks are usually for commands/labels.. "mungers" should probably stay double quoted

File: contributors/devel/automation.md [hunk 5]
```diff
@@ -38,37 +38,33 @@ The status of the submit-queue is [online.](http://submit-queue.k8s.io/)
...
 If the PR has the `retest-not-required` label, it is simply merged. If the PR does
-not have this label the e2e, unit/integration, and node  tests are re-run. If these
-tests pass a second time, the PR will be merged as long as the `critical builds` are
+not have this label, the [Required Github CI Tests](https://github.com/kubernetes/test-infra/blob/master/mungegithub/submit-queue/deployment/kubernetes/configmap.yaml#L15) 
+are re-run. If these tests pass a second time, the PR will be merged as long as the `critical builds` are
 green when this PR finishes retesting.
 
 ## Github Munger
 
 We run [github "mungers"](https://github.com/kubernetes/contrib/tree/master/mungegithub).
```
Comment (cblecker) [inline contributors/devel/automation.md]: This doesn't talk about prow and the difference between mungegithub and prow.

File: contributors/devel/pull-requests.md [hunk 6]
```diff
@@ -108,29 +108,29 @@ Here's the process the PR goes through on its way from submission to merging:
...
 
 1. The PR enters the merge queue ([http://submit-queue.k8s.io](http://submit-queue.k8s.io))
-1. The merge queue triggers a test re-run with the comment `@k8s-bot test this`
-    1. Author has signed the CLA (`cla: yes` label added to PR)
+1. The merge queue triggers a test re-run with the comment `/test all [submit-queue is verifying that this PR is safe to merge]`
+    1. Author has signed the CLA (`cncf-cla: yes` label added to PR)
     1. No changes made since last `lgtm` label applied
 1. If tests fail, resolve issues by pushing edits to your PR branch
-1. If the failure is a flake, a member can comment `@k8s-bot [e2e|unit] test this issue: #<flake issue>`
+1. If the failure is a flake, anyone can comment `/retest`
```
Comment (cblecker) [inline contributors/devel/pull-requests.md]: This isn't true if the PR hasn't been marked okay (e.g. a single test run was started with `/test all` or by `/lgtm`, without the PR itself being marked trusted).

**Comment by bgrant0607** [comment]
cc @kubernetes/sig-testing-pr-reviews @kubernetes/sig-contributor-experience-pr-reviews

File: contributors/devel/automation.md (hunk 5 above)
Comment (spxtr) [reply contributors/devel/automation.md]: Also the link is wrong.

**Comment by xiangpengzhao** [comment]
@cblecker Most of the review comments addressed. PTAL. Thanks!

Please note that I didn't address below comment since I think this is a somehow big topic and I'm not sure how to talk about it now :) Hope other folks can do it in the near future.

>This doesn't talk about prow and the difference between mungegithub and prow.

File: contributors/devel/automation.md [hunk 7]
```diff
@@ -91,24 +83,22 @@ days.
 
 We also run a robotic PR builder that attempts to run tests for each PR.
 
-Before a PR from an unknown user is run, the PR builder bot (`k8s-bot`) asks to
-a message from a contributor that a PR is "ok to test", the contributor replies
-with that message.  ("please" is optional, but remember to treat your robots with
-kindness...)
+Before a PR from an unknown user is run, the PR builder bot (`k8s-ci-robot`) asks to
+a message from a kubernetes member that a PR is `ok-to-test`, the member replies
```
Comment (cblecker) [inline contributors/devel/automation.md]: "kubernetes member that a PR is safe to test. If it is, the member can reply with the `/ok-to-test` command on a single line to begin CI testing."

File: contributors/devel/automation.md [hunk 8]
```diff
@@ -38,37 +38,29 @@ The status of the submit-queue is [online.](http://submit-queue.k8s.io/)
...
+  * All of the required github CI tests must be green. The current list of tests are on the [MERGE REQUIREMENTS tab, at this link](https://submit-queue.k8s.io/#/info)
   * The PR cannot have any prohibited future milestones (such as a v1.5 milestone during v1.4 code freeze)
-  * The PR must have the "lgtm" label. The "lgtm" label is automatically applied
-    following a review comment consisting of only "LGTM" (case-insensitive)
-  * The PR must not have been updated since the "lgtm" label was applied
-  * The PR must not have the "do-not-merge" label
+  * The PR must have the `lgtm` label. The `lgtm` label is automatically applied
+    following a review comment consisting of `/lgtm` (case-insensitive) in a single line
+  * The PR must not have been updated since the `lgtm` label was applied
+  * The PR must have the `approved` label. The `approved` label is automatically applied after all approvers have approved the PR
```
Comment (cblecker) [inline contributors/devel/automation.md]: nit: line length should wrap at 80 chars

File: contributors/devel/automation.md [hunk 9]
```diff
@@ -38,37 +38,29 @@ The status of the submit-queue is [online.](http://submit-queue.k8s.io/)
...
-  * The PR must have the "lgtm" label. The "lgtm" label is automatically applied
-    following a review comment consisting of only "LGTM" (case-insensitive)
-  * The PR must not have been updated since the "lgtm" label was applied
-  * The PR must not have the "do-not-merge" label
+  * The PR must have the `lgtm` label. The `lgtm` label is automatically applied
+    following a review comment consisting of `/lgtm` (case-insensitive) in a single line
+  * The PR must not have been updated since the `lgtm` label was applied
+  * The PR must have the `approved` label. The `approved` label is automatically applied after all approvers have approved the PR 
+    by a comment consisting of `/approve` (case-insensitive) in a single line
+  * The PR must not have the `do-not-merge` label or any of other `do-not-merge/*` labels, e.g., `do-not-merge/release-note-label-needed`
```
Comment (cblecker) [inline contributors/devel/automation.md]: nit: line length should wrap at 80 chars

File: contributors/devel/automation.md [hunk 10]
```diff
@@ -91,24 +83,22 @@ days.
...
 
 PRs should only need to be manually re-tested if you believe there was a flake
-during the original test. All flakes should be filed as an
-[issue](https://github.com/kubernetes/kubernetes/issues?q=is%3Aopen+is%3Aissue+label%3Akind%2Fflake).
-Once you find or file a flake a contributor (this may be you!) should request
-a retest with "@k8s-bot test this issue: #NNNNN", where NNNNN is replaced with
-the issue number you found or filed.
+during the original test. It would be good to file flakes as an
+[issue](https://github.com/kubernetes/kubernetes/issues?q=is%3Aopen+is%3Aissue+label%3Akind%2Fflake). 
+k8s-ci-robot will comment to tell you which test(s) failed and how to re-test.
```
Comment (cblecker) [inline contributors/devel/automation.md]: nit: `@k8s-ci-robot`

File: contributors/devel/pull-requests.md [hunk 11]
```diff
@@ -99,7 +99,7 @@ Now that your release notes are in shape, let's look at how the PR gets tested a
 
 ## The Testing and Merge Workflow
 
-The Kubernetes merge workflow uses comments to run tests and labels for merging PRs. NOTE: For pull requests that are in progress but not ready for review, prefix the PR title with "WIP" and track any remaining TODOs in a checklist in the pull request description.
+The Kubernetes merge workflow uses comments to run tests and labels for merging PRs. NOTE: For pull requests that are in progress but not ready for review, prefix the PR title with `WIP` and track any remaining TODOs in a checklist in the pull request description.
```
Comment (cblecker) [inline contributors/devel/pull-requests.md]: nit: 80 characters. also I think it's `[WIP]` isn't it?

File: contributors/devel/pull-requests.md [hunk 12]
```diff
@@ -99,12 +100,15 @@ Now that your release notes are in shape, let's look at how the PR gets tested a
 
 ## The Testing and Merge Workflow
 
-The Kubernetes merge workflow uses comments to run tests and labels for merging PRs. NOTE: For pull requests that are in progress but not ready for review, prefix the PR title with `WIP` and track any remaining TODOs in a checklist in the pull request description.
+The Kubernetes merge workflow uses comments to run tests and labels for merging PRs.
+NOTE: For pull requests that are in progress but not ready for review,
+prefix the PR title with `WIP` or `[WIP]` and track any remaining TODOs
```
Comment (xiangpengzhao) [inline contributors/devel/pull-requests.md]: >also I think it's [WIP] isn't it?

@cblecker we allow prefix with both `WIP` and `[WIP]`.

https://github.com/kubernetes/test-infra/blob/master/prow/plugins/wip/wip-label.go#L40

**Comment by xiangpengzhao** [comment]
@cblecker PTAL. Thanks!

**Comment by xiangpengzhao** [comment]
@cblecker squashed :)

**Comment by spiffxp** [comment]
Sorry... didn't really complete my thought above. Linking directly to the submit queue's merge requirements would be more ideal.  The docs are generated based on which flags are enabled/disabled, and they're close to the code that implements the logic they describe, so they're more likely to be up to date.  

That said, only a few repos have a submit queue on them, and in ways I can't articulate right now, they submit queue's docs appear less clear than the docs here.  So there's room for improvement over there before we just link directly to eg: https://community.submit-queue.k8s.io/#/info

**Comment by xiangpengzhao** [comment]
@spiffxp https://github.com/kubernetes/community/pull/1110 sent. PTAL. Thanks!

**Pull request merged**